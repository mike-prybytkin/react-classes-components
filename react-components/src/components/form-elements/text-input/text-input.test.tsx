import React from 'react';
import { render, screen } from '@testing-library/react';
import TextInput from './text-input';
import mockText from 'mocks/text';

const setUp = () =>
  render(
    <TextInput
      labelType={mockText.labelUserName}
      inputNameRef={undefined}
      placeholderText={mockText.placeholderUserName}
      minNameLength="2"
      maxNameLength="12"
      required={true}
    />
  );

describe('TextInput component', () => {
  it('should correctly render input', () => {
    setUp();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
  });

  it('should correctly render placeholder', () => {
    setUp();
    expect(
      screen.getByPlaceholderText(new RegExp(mockText.placeholderUserName, 'i'))
    ).toBeInTheDocument();
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
