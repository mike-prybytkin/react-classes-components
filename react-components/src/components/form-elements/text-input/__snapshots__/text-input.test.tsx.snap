// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component should create snapshot 1`] = `
<DocumentFragment>
  <label
    class="user-name__label"
  >
    Name:
    <input
      class="user-name__input"
      maxlength="12"
      minlength="2"
      placeholder="Your Name"
      required=""
      type="text"
    />
  </label>
</DocumentFragment>
`;
